### PR TITLE
Allow opt-in for outline/shadow on focused elements

### LINKED ISSUE #5575: It's currently not possible to add a focus effect (like box-shadow) to a link
### Description

When I add a _focus style including a shadow to a link, I expect that style to be applied

### Link to Reproduction

https://codesandbox.io/s/dazzling-haibt-vnr6u?file=/src/index.tsx

### Steps to reproduce

1. Add a link with a shadow
2. Add the shadow to the _focus styles, too
2. Click the link but make sure the link stays on page (this happens a lot in a Next.js app)
3. The shadow disappears, and there is no way to keep it

### Chakra UI Version

1.8.3

### Browser

_No response_

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

Add `:not(.always-show-focus)` or something to the CSS reset and this is fixed immediately! Document the ability to do this, too. Perhaps generally the focus ring should be documented separately? I'm happy to write that if you want.

### PR DESCRIPTION
**Pull request by tomhicks**
Closes #5575
## 📝 Description

This will need to be documented, but ultimately this allows users to override the automatic focus style overrides applied when using focus-visible.

The focus-visible documentation suggests adding a get-out like this, so Chakra really needs to give us one!

## ⛳️ Current behavior (updates)

When using the focus-visible package, it is impossible to apply certain styles to certain focussed elements. For example, adding a shadow to a link when focussed is currently not possible.

## 🚀 New behavior

By adding a special `data-` attribute, we allow developers to opt-out of this override in certain cases.

## 💣 Is this a breaking change (Yes/No):

No

## 📝 Additional Information

The focus-visible docs are slightly confusing on this. It says first:

> We suggest that users selectively disable the default focus style by selecting for the case when the polyfill is loaded and .focus-visible is not applied to the element:

with this example

```css
.js-focus-visible :focus:not(.focus-visible) {
  outline: none;
}
```

Then they say:

> Alternatively, if you're using a framework which overwrites your classes ([#179](https://github.com/WICG/focus-visible/issues/179)), you can rely on the data-js-focus-visible and data-focus-visible-added attributes.

And give this example, which is what Chakra already has:

```css
[data-js-focus-visible] :focus:not([data-focus-visible-added]) {
  outline: none;
}
```

The problem is that you can't just add `data-focus-visible-added` or whatever to your element, because the focus-visible package goes around adding and removing that attribute - it can work for one render, then the attribute is removed by the library, and it's broken again.

I think their docs are wrong, or possibly just confusing. You do need an extra opt-out selector.

### CODE REVIEW

File: packages/css-reset/src/css-reset.tsx [hunk 1]
```diff
@@ -271,7 +271,7 @@ export const CSSReset = () => (
         height: auto;
       }
 
-      [data-js-focus-visible] :focus:not([data-focus-visible-added]) {
+      [data-js-focus-visible] :focus:not([data-focus-visible-added]):not([data-chakra-allow-focus-highlights]) {
```
[the remark this answers is not in the document]
Comment (tomhicks) [reply packages/css-reset/src/css-reset.tsx:274]: This isn't correct. This has broken the selector for the focus-visible plugin, so the reason you're seeing it "work" is that focus-visible is not working at all.

You need to add the data-chakra-allow-focus-highlights attribute to opt-out of the shadow-hiding behaviour.

Hopefully this shows it more clearly:
https://codesandbox.io/s/zealous-nobel-gw1hhh?file=/src/index.tsx

**Review by TimKolberger — CHANGES_REQUESTED** [review]
Thanks for creating this PR @tomhicks 💖 
I added my thoughts as comments.

**Comment by tomhicks** [comment]
@TimKolberger can we get some eyes on this? I've implemented your change of selector as requested.

But this not being in the deployed package is pretty annoying because I still can't add focus styles to buttons.

**Review by TimKolberger — APPROVED** [review]
Thanks for working on this @tomhicks ✨

**Pull request merged**